translations/pt-BR/README.md

```diff
@@ -0,0 +1,323 @@
+
+Vamos agora atualizar as regras do linter para verificar se [tipos de propriedade](https://reactjs.org/docs/typechecking-with-proptypes.html) devem ser definidos em todos os componentes do React. Para definir esta regra linter, abra o arquivo `.eslintrc`. Dentro do objeto `rules` adicione: `"react/prop-types": "warn"`. Suas regras ESLint devem ser:
+
+```javascript
+"rules": {
+   "prettier/prettier": ["warn", { "endOfLine": "auto" }],
+   "react/prop-types": "warn"
+}
+```
+
+> _Nota: os valores possíveis para definir uma regra são "off", "warn" e "error". Quando definido como "warn", o aluno será notificado sobre o problema, mas não será obrigado a resolvê-lo. Definir como "error" exigirá que um aluno resolva esse problema de linter._
+
+Com isso definido, todas as propriedades de entrada para um componente precisarão ser definidas com o nome e o tipo ou o aluno verá um aviso. Você pode então adicionar um prop de título a `Header` e ver sua nova regra em ação:
+
+<img src="https://user-images.githubusercontent.com/82035/198326157-3f6e28b4-d724-4725-b5ed-5331d8c17db7.png" alt="Header component with title prop and linter error" style="width: 600px" />
+
+Para resolver a questão dos tipos de prop neste exemplo, você precisaria importar `PropTypes` e então definir os `propTypes` para `Header`, dando a você:
+
+```javascript
+import React from "react";
+import PropTypes from "prop-types";
+
+const Header = ({ title }) => {
+  return <h1>Educator React Codespaces JS Template - {title}</h1>;
+};
+
+Header.propTypes = {
+  title: PropTypes.string,
+};
+
+export default Header;
+```
+
+### Step 3: Atualize a versão do Node.js
+
+Se você deseja alterar qual versão do Node.js esse projeto está usando, siga essas etapas:
+
+Abra *.devcontainer/devcontainer.json* e substitua a seguinte ação:
+
+```json
+"VARIANT": "16-bullseye"
+```
+
+com a seguinte instrução:
+
+```json
+"VARIANT": "18.9-bullseye"
+```
+
+Essa alteração usará Node.js 18.9 em vez de 16. A lista completa de todas as variantes Node.js disponíveis pode ser encontrada em [hub.docker.com/_/node](https://hub.docker.com/_/node)
+
+Lembrete: Quando você alterar qualquer configuração no json, uma caixa apareecerá após salvar.
+
+Clique em reconstruir. Aguarde até que seu espaço de código recrie o ambiente VS Code.
+
+<br />
+
+## 🤖 Criar uma lição
+
+Este modelo de projeto fornece uma base para você criar uma lição personalizada. Usanto o GitHub Copilot, vamos orientá-lo através da criação de uma lição de  exemplo[Typescript](https://www.typescriptlang.org/)  e um quiz que você pode usar para sua classe nas seguintes etapas
+
+1. [Escreva uma descrição da lição](#step-1-write-a-lesson-description)
```

Comment:
verificar os links, se estão navegando para os trechos corretos neste tutorial

Reply:
Feito > Links checados.

Reply:
os links não estão navegando no markdown. abra o seu arquivo e clique nos títulos para poder o link # correto.
https://github.com/rogergcorrea27/codespaces-teaching-js-template/blob/Translation-to-PT-BR-1/translations/pt-BR/README.md

Por exemplo, o link do passo 1 é #etapa-1-escrever-uma-descrição-para-lição